modules/plugin/mongodb/src/main/java/org/geotools/data/mongodb/AbstractFilterToMongo.java

```diff
@@ -181,11 +184,30 @@ public Object visit(And filter, Object extraData) {
         BasicDBObject output = asDBObject(extraData);
         List<Filter> children = filter.getChildren();
         BasicDBList andList = new BasicDBList();
+        boolean aggregationQuery = false;
```

Comment:
It is not clear to me what's happening here. Would this code work for any sort of and filter that contains a DWithin? What if the situation is more complex, and we have a not-around this or, or some other sort of logic filter nesting? Also, what about "or"? I have the feeling this is a "special case" fix that would not work in general, can you explain more what is happening here?

Reply:
Indeed I had tunnel vision and did not check other possibilities. This way only "and" and "dwithin" filters would support aggregation, so I guess in this case we should extend AbstractFilterToMongo with something like AggregationFilterToMongo and use that in case of aggregation.

Reply:
Removed this due to not using aggregation anymore